src/ActionServer_FJT.c

```diff
@@ -482,6 +482,142 @@ void Ros_ActionServer_FJT_ProcessFeedback()
     }
 }
 
+static STATUS Ros_ActionServer_FJT_Parse_GoalPosTolerances(
+    control_msgs__msg__JointTolerance__Sequence const* const goal_joint_tolerances /* in */,
+    rosidl_runtime_c__String__Sequence const* const joint_names /* in */,
+    double* posTolerances /* in/out */, size_t posTolerances_len /* in */)
+{
+    if (goal_joint_tolerances == NULL || joint_names == NULL || posTolerances == NULL)
+    {
+        Ros_Debug_BroadcastMsg("%s: one or more inputs NULL, abort", __func__);
+        return -1;
+    }
+
+    //always configure defaults
+    for (int i = 0; i < posTolerances_len; ++i)
+        posTolerances[i] = DEFAULT_FJT_GOAL_POSITION_TOLERANCE;
+
+    //if caller hasn't passed any JointTolerances, set all entries to default
+    if (goal_joint_tolerances->size == 0)
+    {
+        Ros_Debug_BroadcastMsg("%s: no joint tolerances specified, returning %d defaults",
+            __func__, posTolerances_len);
+        return OK;
+    }
+
+    //it's OK if we are passed more JointTolerance instances than we have joints,
+    //as technically a client could send duplicate tolerance specs (ie: multiple
```

Comment:
Is this common practice? It seems to me that this is a bad idea. If I accidently duplicate a joint, then I would be very confused when my intended tolerance isn't applied.

Reply:
not common practice per se. I just didn't feel like making it an issue.

We could upgrade it to one though, and return a specific error code if duplicated joint names are passed as part of the goal.